### PR TITLE
ruff and pre-commit config files

### PR DESCRIPTION
**Pull request by ArshSaja**
## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Checklist

- [ ] I have run `ruff check` and `ruff format` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Review by eirikurj — APPROVED** [review]

**Review by eirikurj — CHANGES_REQUESTED** [review]
There are still some issue with the formatting and linting checks. Also the tests are failing with `cgio_get_label:ADF 17: Internal error: Disk boundary tag bad.` Looks like the cgns files are touched in this PR, why is this? I am assuming this is done by the pre-commit linting? Dont think we should be touching any binary files.

File: cgnsutilities/cgnsutilities.py [hunk 1]
```diff
@@ -2883,9 +2905,31 @@ def simpleCart(xMin, xMax, dh, hExtra, nExtra, sym, mgcycle, outFile):
 # -----------------------------------------------------------------
 # These functions perform operations that return new 'Grid' objects
 # -----------------------------------------------------------------
+def _getBocoDataSetArray(inFile, iBlock, iBoco, iBocoDataSet, BCDataType, iDir):
```
Comment (eirikurj) [inline cgnsutilities/cgnsutilities.py:2908]: Only commenting here, but this applies elsewhere as well. Couple of things,
1. Even though its small, I dont think we should make code changes like this (variable renaming and moving code) in a PR meant for linting (unless its relevant to the linting). Makes blaming less effective, etc. Can you please separate this (and related changes) out into a separate PR?
2. To be consistent with current naming scheme (also too close to class naming), change `BCDataType` to `bcDataType` (and elsewhere).

File: src/cgns_utils.pyf [hunk 2]
```diff
@@ -51,25 +51,25 @@ python module libcgns_utils ! in
             integer intent(out) :: ndirichletarrays
             integer intent(out) :: nneumannarrays
         end subroutine getbcdatasetinfo
-        subroutine getbcdataarrayinfo(cg,iblock,ibc,ibcdataset,idataarr,flagdirneu,dataarrayname,datatype,ndatadimensions,datadimensionvector) ! in :libcgns_utils:cgns_utilities.F90
+        subroutine getbcdataarrayinfo(cg,iblock,ibc,ibcdataset,idataarr,BCDataType,dataarrayname,datatype,ndatadimensions,datadimensionvector) ! in :libcgns_utils:cgns_utilities.F90
```
Comment (eirikurj) [inline src/cgns_utils.pyf:54]: f2py uses all lower case by default and the signature file is case-sensitive so the name needs to be all lower case

**Review by eirikurj — CHANGES_REQUESTED** [review]

File: cgnsutilities/cgnsutilities.py (hunk 1 above)
Comment (A-CGray) [reply cgnsutilities/cgnsutilities.py:2908]: Good point, moved this change to #79

**Review by A-CGray — APPROVED** [review]

File: cgnsutilities/cgnsutilities.py (hunk 1 above)
Comment (eirikurj) [reply cgnsutilities/cgnsutilities.py:2908]: Just to check, do you plan to revert this relocation of this function and the changes as well?

Comment (A-CGray) [reply cgnsutilities/cgnsutilities.py:2908]: No, the relocation was necessary to meet one of the ruff linting rules

Comment (eirikurj) [reply cgnsutilities/cgnsutilities.py:2908]: ok sounds good, do you remember what rule this was ?

**Review by eirikurj — APPROVED** [review]
